pgzx-update-0.2.0.mdx

```diff
@@ -0,0 +1,406 @@
+the error and return `PGErrorStack` to indicate that a Postgres error was
+reported. The Zig error will be returned by the hello world function. The
+wrapper generated by `PG_FUNCTION_V1` will then do a longjump to pass the error
+back to PostgresSQL.
+
+Exporting a number of functions can become a bit tedious when you want to
+export multiple functions. In that case you might want to use `PG_EXPORT`.
+`PG_EXPORT` will export all public functions from the struct given:
+
+```zig
+comptime {
+    pgzx.PG_EXPORT(sqlfns);
+}
+
+const sqlfns = struct {
+    pub fn hello_world(name: ?[:0]const u8) ![:0]const u8 {
+        return if (name) |n|
+            try std.fmt.allocPrintZ(pgzx.mem.PGCurrentContextAllocator, "Hello, {s}!", .{n})
+        else
+            "Hello World";
+    }
+};
+```
+
+We have added a sample extension called
+[`sqlfns`](https://github.com/xataio/pgzx/tree/main/examples/sqlfns) that
+demonstrates multiple ways on how you can write a SQL function in Zig. The
+sample extension documents different ways on how to export a function, accept
+arguments and return values.
+
+## Error logging API overhaul (`pgzx.elog`)
+
+We still support the helper functions `elog.Error`, `elog.Warning`,
+`elog.Notice` and friends. We also introduced `WithCause` variants like:
+
+```zig
+pub fn WarningWithCause(src: SourceLocation, cause: anyerror, comptime fmt: []const u8, args: anytype) void;
+```
+
+The `WithCause` variants will check the `cause` argument. If the error code is
+`PGErrorStack` we emit the captured error report and emit the new report. If it
+is any other Zig error code we append it to the message. These utilities are
+mostly for convenience and nice to have when adding logs for debugging
+purposes.
+
+The underlying error reporting API has been rewritten to follow the C API more closely. For example:
+
+```zig
+    elog.ereport(@src(), .Error, .{
+        elog.errcode(pg.ERRCODE_UNDEFINED_OBJECT),
+        elog.errmsg("Foreign schema \"{s}\" does not exist", .{foreign_schema_name}),
+    });
+```
+
+Originally we've been considering to use the builder pattern for the elog API,
+but we found it a little inflexible when combinig the builder with the C-API.
+Plus we had been forced to allocate buffers for messages in advance, even if
+the PostgreSQL would have discarded the message.
+
+The new API works more similar to the C-API. In the example given `ereport`
+will check if the message will be emitted or discarded by calling `errstart`.
+The `errcode` and `errmsg` functions will only be executed if `errstart`
+returns true.
+
+How does this work? Well, we use a mix of comptime and a pattern known to the
+[Go](https://go.dev/) community as "Functional Options". In essence in the
+functional options pattern a function accepts a variable number of arguments
+where each argument implements a common interface. This can be a function or an
+actual Go interface. The receiver might establishe an internal 'context' and
```

Comment:
I got a bit confused by:

> This can be a function or an actual Go interface.

Is it about the arguments?

Reply:
Yeah, it is about the argument.